src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,318 @@
+AD and IPA.
+
+Another difference is that with the current IPA feature, the first and second
+factor can also be used for doing a direct LDAP bind. This will not be possible
+with the new passkey scheme. Direct LDAP bind support will not be in the scope
+of this implementation.
+
+Implementation details
+----------------------
+
+The implementation will loosely follow existing smartcard authentication in
+SSSD. On top of that, and as mentioned before, the information will be stored
+in two places: in the LDAP server and on the passkey. The server will contain
+the public user credentials, including the user key mapping data, while the
+passkey device will store the private key necessary to authenticate the user.
+
+Key registration
+****************
+The first step is to register the user with a key. SSSD will provide a helper
+utility to handle the passkeys, and it can be used to create the key mapping
+string (similar to pamu2fcfg). Instead of calling the helper utility from the
+command line the system administrator would use 
+`sssctl <https://docs.pagure.org/sssd.sssd/design_pages/sssctl.html>`__. This
+would enable to unify the user experience. Then, the output of this command can
+be set to the LDAP server in a dedicated attribute. For AD we could use the
+`altSecurityIdentities <https://learn.microsoft.com/en-us/windows/win32/adschema/a-altsecurityidentities>`__
+with a prefix (i.e. passkey). For FreeIPA we would use a new attribute called
+ipapasskey from the ipapasskeyuser objectclass. For any other LDAP server we
+would use the passkey attribute.
+
+The format for
+`server-side credentials <https://developers.yubico.com/WebAuthn/WebAuthn_Developer_Guide/Resident_Keys.html>`__
+for the key mapping will be
+``passkey:credentialId,pemPublicKey``.
+
+The format for
+`discoverable credentials <https://developers.yubico.com/WebAuthn/WebAuthn_Developer_Guide/Resident_Keys.html>`__ for the key mapping will be
+``passkey:credentialId,pemPublicKey,userId``.
+
+IPA process
++++++++++++
+FreeIPA would provide a direct way to register the passkey attributes to the
+user entry. This can be done with a command of IPA CLI or directly in the
+WebUI. The initial implementation would focus on the IPA CLI integration. As
+mentioned before the user would call sssctl to register the key. The output of
+this command would be used to populate the passkey attribute in FreeIPA.
+
+On top of that, FreeIPA can take advantage of libfido2 to communicate with the
+device to perform this whole process.
+
+AD (and any other LDAP server) process
+++++++++++++++++++++++++++++++++++++++
+The users, or the system administrator, would call sssctl. The user can
+manually add this information to their AD object using AD's 'User and
+Computers' utility, or any other attribute in case of another LDAP server.
+
+LDAP attributes
+***************
+Newly added IPA ``ipapasskey`` LDAP attribute will be added to
```

Comment:
Please rephrase as factual statement. E.g. 'FreeIPA ID provider will use LDAP attribute ``ipaPassKey`` to retrieve the passkey mapping data ...".